src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AwsSdk/LambdaInvokeRequestHandler.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Collections.Concurrent;
+using System;
+using System.Threading.Tasks;
+using NewRelic.Agent.Api;
+using NewRelic.Agent.Api.Experimental;
+using NewRelic.Agent.Extensions.Providers.Wrapper;
+using NewRelic.Reflection;
+using NewRelic.Agent.Extensions.Helpers;
+
+namespace NewRelic.Providers.Wrapper.AwsSdk
+{
+    internal static class LambdaInvokeRequestHandler
+    {
+        private static ConcurrentDictionary<Type, Func<object, object>> _getResultFromGenericTask = new();
+        private static ConcurrentDictionary<string, string> _arnCache = new ConcurrentDictionary<string, string>();
+
+        private static object GetTaskResult(object task)
+        {
+            if (((Task)task).IsFaulted)
+            {
+                return null;
+            }
+
+            var getResponse = _getResultFromGenericTask.GetOrAdd(task.GetType(), t => VisibilityBypasser.Instance.GeneratePropertyAccessor<object>(t, "Result"));
```

Comment:
If you are using dynamic anyways, do you need to use the concurrent dictionary and visibility bypasser combination here?

Reply:
That's the pattern we use elsewhere, though now that I look at it I'm not sure why. I can simplify it.